Assets/PatchKit Patcher/Scripts/Debug/DebugMenu.cs

```diff
@@ -0,0 +1,216 @@
+        }
+
+        void OnGUI()
+        {
+            if (Event.current.Equals(Event.KeyboardEvent(KeyCode.D.ToString())))
+            {
+                _show = !_show;
+                _graphicRaycaster.enabled = !_graphicRaycaster.enabled;
+            }
+
+            if (_show)
+            {
+                GUI.DrawTexture(_debugMenu, Texture2D.whiteTexture);
+                GUI.Window(0, _debugMenu, Menu, "Debug Menu");
+            }
+            else if (_showPopUp)
+            {
+                GUI.DrawTexture(_popUp, Texture2D.whiteTexture);
+                GUI.Window(1, _popUp, PopUp, "Information");
+            }
+        }
+
+        void Menu(int i)
+        {
+            if (GUILayout.Button("Open Patcher log file"))
+            {
+                OpenPatcherLogFile();
+            }
+
+            if (GUILayout.Button("Open Patcher log file location"))
+            {
+                OpenPatcherLogFileLocation();
+            }
+
+            if (GUILayout.Button("Open Launcher log file"))
+            {
+                OpenLauncherLogFile();
+            }
+
+            if (GUILayout.Button("Open Launcher log file location"))
+            {
+                OpenLauncherLogFileLocation();
+            }
+
+            if (Patcher.Instance.IsAppInstalled.Value)
+            {
+                if (GUILayout.Button("Verify all app files"))
+                {
+                    VerifyAllAppFiles();
+                }
+
+                if (GUILayout.Button("Remove all app files"))
+                {
+                    RemoveAllAppFiles();
+                }
+            }
+        }
+
+        private void VerifyAllAppFiles()
+        {
+            SetUserDecision(Patcher.UserDecision.VerifyFiles);
+        }
+
+        private void RemoveAllAppFiles()
+        {
+            SetUserDecision(Patcher.UserDecision.UninstallApp);
+        }
+
+        private void SetUserDecision(Patcher.UserDecision userDecision)
+        {
+            while (Patcher.Instance.State.Value != PatcherState.WaitingForUserDecision)
+            {
+            }
+
+            Patcher.Instance.SetUserDecision(userDecision);
+
+            _show = false;
+            _graphicRaycaster.enabled = true;
+        }
+
+        void Open(string path)
+        {
+            _show = false;
+            try
+            {
+                Process.Start(path);
+                _graphicRaycaster.enabled = true;
+            }
+            catch (Exception e)
+            {
+                _message = string.Format("The directory/file cannot be found: {0}", path);
+                DebugLogger.LogError(_message);
+                _showPopUp = true;
+            }
+        }
+
+        private void OpenLauncherLogFileLocation()
+        {
+#if UNITY_EDITOR
+            _message = "Access to Launcher in the editor is not possible";
+            _show = false;
+            _showPopUp = true;
```

Comment:
Please introduce a new method `OpenPopup(string popupMessage)` and use it here.

Reply:
done